sql/src/test/java/io/crate/integrationtests/KillIntegrationTest.java

```diff
@@ -96,6 +104,69 @@ public void run() {
         }
     }
 
+    private void assertCancelledByJobId(final String statement, @Nullable final Object[] params) throws Exception {
+        ExecutorService executor = Executors.newSingleThreadExecutor();
+        final AtomicReference<Throwable> thrown = new AtomicReference<>();
+        try {
+            executor.submit(new Runnable() {
+                @Override
+                public void run() {
+                    while (thrown.get() == null) {
+                        try {
+                            execute(statement, params);
+                        } catch (Throwable e) {
+                            Throwable unwrapped = Exceptions.unwrap(e);
+                            thrown.compareAndSet(null, unwrapped);
+                        }
+                    }
+                }
+            });
+
+            Thread.sleep(100L);
+
+            SQLResponse logResponse = execute("select * from sys.jobs where stmt = ?", new Object[]{statement});
+            String jobId = logResponse.rows()[0][0].toString();
+            SQLResponse response = execute("kill ?", new Object[] {jobId});
+
+            executor.shutdown();
+            executor.awaitTermination(5L, TimeUnit.SECONDS);
+
+            Throwable exception = thrown.get();
+            if (exception != null) {
+                assertThat(exception, instanceOf(SQLActionException.class));
+                assertThat(((SQLActionException)exception).stackTrace(),  anyOf(
+                        containsString("Job killed by user"), // CancellationException
+                        containsString("JobExecutionContext for job"), // ContextMissingException when job execution context not found
+                        containsString("SearchContext for job") // ContextMissingException when search context not found
+                ));
+            }
+        } finally {
+            executor.shutdownNow();
+        }
+    }
+
+    @Test
+    public void testKillCopyToJobById() throws Exception {
+        String path = temporaryFolder.newFolder().getAbsolutePath();
+        setup.setUpEmployees();
+        assertCancelledByJobId("copy employees to directory ?", new Object[]{path});
```

Comment:
do we need so many ById kill integration tests? what does each of those test that is differemt from kill all?